arangod/RestHandler/RestTestHandler.cpp

```diff
@@ -0,0 +1,163 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2016 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "RestTestHandler.h"
+
+#include "GeneralServer/GeneralServer.h"
+#include "GeneralServer/GeneralServerFeature.h"
+#include "GeneralServer/RestHandlerFactory.h"
+#include "Logger/Logger.h"
+#include "Rest/HttpRequest.h"
+#include "Utils/ExecContext.h"
+#include "Cluster/ResultT.h"
+#include "Scheduler/SchedulerFeature.h"
+
+using namespace arangodb;
+using namespace arangodb::basics;
+using namespace arangodb::rest;
+
+RestTestHandler::RestTestHandler(GeneralRequest* request,
+                                   GeneralResponse* response)
+    : RestVocbaseBaseHandler(request, response) {}
+
+RestTestHandler::~RestTestHandler() {}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief was docuBlock JSF_batch_processing
+////////////////////////////////////////////////////////////////////////////////
+
+ResultT<RequestLane> RestTestHandler::requestLaneFromString(const std::string &str)
+{
+  #define LANE_ENTRY(s) {#s, RequestLane:: s},
+  static std::map<std::string, RequestLane> lanes = {
```

Comment:
works, but in general static things should not be declared inside functions, because the compiler needs to put extra `is-this-variable-already-initialized` checks around it. Rather put it into `namespace {}` in the beginning of this file, make it const, and initialize it statically before `main`.